docs/index.rst

```diff
@@ -1,260 +1,427 @@
 Confit: Painless Configuration
 ==============================
 
-`Confit`_ (*con-FEE*) is a straightforward, full-featured configuration system
-for Python.
+`Confit`_ (*con-FEE*) is a straightforward, full-featured configuration
+library written in Python.
 
 .. _Confit: https://github.com/sampsyo/confit
 
+When to Use Confit
+------------------
 
-Using Confit
-------------
+Confit abstracts the process of compiling a runtime configuration. In this
+context, a configuration holds values that would otherwise be hard coded
+into your application or library. Common examples include:
 
-Set up your Configuration object, which provides unified access to
-all of your application’s config settings::
+- A host and port for a remote service.
+- A username and password for a remote service.
+- A port number to bind.
+- A timeout in milliseconds.
+- A path to a temporary directory.
+- A log level.
 
-    config = confit.Configuration('MyGreatApp', __name__)
+Configurations are distinct from data, although the line is hard to define.
+In our approximation, an application can run without data, but it cannot run
+without a configuration. A configuration can be packaged and imported, but
+data cannot.
```

Comment:
I like this philosophy statement! I'm not 100% sure what "packaged" or "imported" mean in this context, but I'm willing to believe them.